history/list-facility-history.js

```diff
@@ -0,0 +1,97 @@
+/*
+    This example demonstrates how to get history of elements within facility.
+    
+    It uses 2-legged authentication - this requires athat application is added to facility as service.
+
+    NOTE - This example uses API which is NOT SUPPORTED at the moment:
+        POST https://developer.api.autodesk.com/tandem/v1/modeldata/:modelId/history
+*/
+import { createToken } from '../common/auth.js';
+import { TandemClient } from '../common/tandemClient.js';
+import { ColumnFamilies, QC } from '../common/utils.js';
+
+// update values below according to your environment
+const APS_CLIENT_ID = 'YOUR_CLIENT_ID';
+const APS_CLIENT_SECRET = 'YOUR_CLIENT_SECRET';
+const FACILITY_URN = 'YOUR_FACILITY_URN';
+const OFFSET_DAYS = 10;
+
+async function main() {
+    // STEP 1 - obtain token to authenticate subsequent API calls
+    const token = await createToken(APS_CLIENT_ID,
+        APS_CLIENT_SECRET, 'data:read');
+    const client = new TandemClient(() => {
+        return token;
+    });
+
+    // STEP 2 - get facility
+    const facilityId = FACILITY_URN;
+    const facility = await client.getFacility(facilityId);
+    // STEP 3 - calculate dates for history
+    const now = new Date();
+    const to = now.getTime();
+    const from = new Date(now.getTime() - OFFSET_DAYS * 24 * 60 * 60 * 1000).getTime();
+
+    // STEP 4 - iterate through facility models
+    for (const link of facility.links) {
+        const schema = await client.getModelSchema(link.modelId);
+        // STEP 5 - get history between dates
+        const history = await client.getModelHistoryBetweenDates(link.modelId, from, to, true, false);
+        // extract list of keys and related timestamp (k, t)
+        const entries = history.flatMap((i) => i.k?.map((j) => ( { k: j, t: i.t })))
```

Comment:
I would advise using more descriptive variable names here. At least whenever we can.